app/bundles/CoreBundle/Twig/Extension/ThemesExtension.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CoreBundle\Twig\Extension;
+
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class ThemesExtension extends AbstractExtension
+{
+    public function __construct(private CoreParametersHelper $coreParametersHelper)
+    {
+    }
+
+    public function getFunctions()
+    {
+        return [
+            new TwigFunction('theme_text_on_color', [$this, 'theme_text_on_color']),
+            new TwigFunction('theme_text_on_color__helper', [$this, 'theme_text_on_color__helper']),
+            new TwigFunction('theme_primary_color', [$this, 'theme_primary_color']),
+        ];
+    }
+
+    public function theme_primary_color(): string
```

Comment:
We use camelCase for method naming in the Mautic code base. Let's be consistent.
```suggestion
    public function themePrimaryColor(): string
```